### PR TITLE
restore base operations with favorites

### PR DESCRIPTION
**Pull request by ctapmex**
## Summary
Restore base operations with favorites.
 
## Checklist
- [x] I have followed the [contributing guidelines](https://github.com/FarGroup/FarManager/blob/master/CONTRIBUTING.md).
- [ ] I have discussed this with project maintainers:  <br/>
If not checked, I accept that this work might be rejected in favor of a different grand plan.<br/>

### CODE REVIEW

**Comment by ctapmex** [comment]
> rollback 'remove RestoreScreen usage'

need for long network scan. Later look again, but now without it broke functional

**Comment by alabuzhev** [comment]
Looks good, please also:
- update version.hpp
- update changelog, reflecting yesterday's changes as well.

**Comment by ctapmex** [comment]
update

File: plugins/network/NetFavorites.cpp [hunk 1]
```diff
@@ -2,84 +2,86 @@
...
+		i++;
+	}
+
+	std::wstring machine_name{remote_name.substr(i)};
+	auto const pos = machine_name.find(L'\\');
+	if (pos != std::string::npos)
+	{
+		auto share_name = machine_name.substr(pos + 1);
+		machine_name = machine_name.substr(0, pos);
+		std::use_facet<std::ctype<wchar_t>>(std::locale()).toupper(
```
Comment (alabuzhev) [inline plugins/network/NetFavorites.cpp:42]: This only works with the base ascii character set.
I'd prefer delegating *everything* locale-, case- etc. related to the OS, e.g. via [CharUpperBuffW](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-charupperbuffw).

This is probably not critical in this particular place - machine names are restricted to [A-Z0-9.-] anyway - but still, eventually the code could be copy-pasted elsewhere.

File: plugins/network/NetFavorites.cpp [hunk 2]
```diff
@@ -2,84 +2,86 @@
...
+			&machine_name[0],
+			&machine_name[0] + machine_name.size());
+		result.push_back(machine_name);
+		result.push_back(share_name);
+	}
+	else
+	{
+		std::use_facet<std::ctype<wchar_t>>(std::locale()).toupper(
```
Comment (alabuzhev) [inline plugins/network/NetFavorites.cpp:50]: Same

File: plugins/network/NetFavorites.cpp [hunk 3]
```diff
@@ -2,84 +2,86 @@
...
+	}
+
+	std::wstring machine_name{remote_name.substr(i)};
+	auto const pos = machine_name.find(L'\\');
+	if (pos != std::string::npos)
+	{
+		auto share_name = machine_name.substr(pos + 1);
+		machine_name = machine_name.substr(0, pos);
+		std::use_facet<std::ctype<wchar_t>>(std::locale()).toupper(
+			&machine_name[0],
```
Comment (alabuzhev) [inline plugins/network/NetFavorites.cpp:43]: `.data()`?

File: plugins/network/NetFavorites.cpp [hunk 4]
```diff
@@ -2,84 +2,86 @@
...
-					lstrcat(szSrc, szSubKey);
-					pList->Push(tmp);
-				}
-				else
-				{
-					szSrc[0] = L'\\';
-					szSrc[1] = L'\\';
-					szSrc[2] = 0;
-					lstrcat(szSrc, szSubKey);
+					wchar_t szSrc[MAX_PATH] = L"\\\\\0";
```
Comment (alabuzhev) [inline plugins/network/NetFavorites.cpp:82]: Why \0 here? The literal is null-terminated anyway.

File: plugins/network/NetFavorites.cpp [hunk 5]
```diff
@@ -93,701 +95,139 @@ BOOL GetFavorites(LPNETRESOURCE pNR, NetResourceList* pList)
...
-			if (p)
-			{
-				lstrcpy(res, GetMsg(MFavorites));
-				lstrcat(res, p);
-				p = res;
-			}
-
-			nr.lpRemoteName = p;
-			NetResourceList::CopyNetResource(*lpParent, nr);
+			level_root = settings.CreateSubKey(level_root, piece.c_str());
```
Comment (alabuzhev) [inline plugins/network/NetFavorites.cpp:154]: Why do we *create* subkeys in *Read*FavoriteItem?

**Pull request merged**